csv_generator/consumers/consumer.py

```diff
@@ -0,0 +1,56 @@
+import csv
+import os
+from typing import List
+
+from bs4 import BeautifulSoup
+
+
+class BaseXMLConsumer:
+    delimiter = ','
+    headers = []
+    base_file_name = 'base_consumer.csv'
+    _output_file = ''
+
+    def __init__(self, create_date: str, zip_file_name: str, output_dir: str = '') -> None:
+        self.create_date = create_date
+        self.zip_file_name = zip_file_name
+        self.output_dir = output_dir
+        self._delete_old_output_file()
+        self._write_row(self.headers)
+
+    def _delete_old_output_file(self):
+        if os.path.isfile(self.output_file):
+            os.remove(self.output_file)
+
+    @staticmethod
+    def get_contents(soup: BeautifulSoup, target_ele: str) -> str:
+        try:
+            return soup.find(target_ele).contents[0]
```

Comment:
I was never quite sure what _BeautifulSoup_ adds for XML processing over _lxml_. Is _contents_ returning the the text of child as well rather than just _text_ in _lxml_?

Reply:
That's right.

Reply:
*BeautifulSoup* does seem to be quite slow. Just the parsing, I am getting about 5.5 it/s whereas lxml's etree.parse is about 200 it/s (as per tqdm). That mainly shows when processing a year's worth of zip. I'd be tempted to go to lxml directly (perhaps as a separate PR).